handlers/transcode.go

```diff
@@ -72,9 +68,179 @@ func (d *CatalystAPIHandlersCollection) TranscodeSegment() httprouter.Handle {
-			errors.WriteHTTPInternalServerError(w, "error send transcode error", err)
+		if err := RunTranscodeProcess(d.MistClient, transcodeRequest); err != nil {
+			errors.WriteHTTPInternalServerError(w, "Error running Transcode process", err)
 		}
-		_, _ = io.WriteString(w, "OK") // TODO later
 	}
 }
+
+// stream from a source to a destination buffer while also printing
+func streamOutput(src io.Reader, dst *bytes.Buffer, out io.Writer) error {
+	mw := io.MultiWriter(dst, out)
+	s := bufio.NewReader(src)
+	for {
+		var line []byte
+		line, err := s.ReadSlice('\n')
+		if err == io.EOF && len(line) == 0 {
+			break
+		}
+		if err == io.EOF {
+			return fmt.Errorf("Improper termination: %v", line)
+		}
+		if err != nil {
+			return err
+		}
+		_, err = mw.Write(line)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// RunTranscodeProcess starts `MistLivepeeerProc` as a subprocess to transcode inputStream into renditionsStream.
+func RunTranscodeProcess(mistClient clients.MistAPIClient, request TranscodeSegmentRequest) error {
+	inputStream, renditionsStream := config.GenerateStreamNames()
+	if err := mistClient.AddStream(inputStream, request.SourceFile); err != nil {
+		return fmt.Errorf("error adding stream to Mist: %s", err)
+	}
+
+	configPayload, err := json.Marshal(configForSubprocess(request, inputStream, renditionsStream))
+	if err != nil {
+		return fmt.Errorf("ProcLivepeerConfig json encode: %s", err)
+	}
+	args := string(configPayload)
+
+	transcodeCommand := exec.Command(config.PathMistProcLivepeer, args, "--debug", "8", "--kickoff")
+
+	var stdout, stderr bytes.Buffer
+	stderrPipe, err := transcodeCommand.StderrPipe()
+	if err != nil {
+		return fmt.Errorf("Failed to open stderr pipe: %s", err)
+	}
+	stdoutPipe, err := transcodeCommand.StdoutPipe()
+	if err != nil {
+		return fmt.Errorf("Failed to open stdout pipe: %s", err)
+	}
+
+	// Start the Transcode Command asynchronously - we call Wait() later in this method
+	fmt.Printf("Starting transcode via: %s\n", transcodeCommand.String())
+	err = transcodeCommand.Start()
+	if err != nil {
+		return fmt.Errorf("Failed to start MistProcLivepeer: %s", err)
+	}
+
+	go func() {
+		if streamOutput(stdoutPipe, &stdout, os.Stdout) != nil {
+			fmt.Errorf("Failed to stream output from stdout")
+		}
+	}()
+	go func() {
+		if streamOutput(stderrPipe, &stderr, os.Stderr) != nil {
+			fmt.Errorf("Failed to stream output from stderr")
+		}
+	}()
+
+	// TODO: remove when Mist code is updated https://github.com/DDVTECH/mistserver/issues/81
+	// Starting SOURCE_PREFIX stream because MistProcLivepeer is unable to start it automatically
+	file, err := os.CreateTemp("", "tmp-*.ts")
+	if err != nil {
+		return fmt.Errorf("error creating temporary file for SOURCE_PREFIX stream: %s", err)
+	}
+
+	if err := mistClient.PushStart(inputStream, file.Name()); err != nil {
+		return fmt.Errorf("PushStart(inputStream): %s", err)
+	}
```

Comment:
With latest mistserver updates, is this step still required?

Reply:
Good point, will test it out after removing it - technically shouldn't be required anymore.